src/hooks/useTransactionStatus.ts

```diff
@@ -1,22 +1,39 @@
 import { useAppSelector } from '@/store'
 import { PendingStatus, selectPendingTxById } from '@/store/pendingTxsSlice'
+import { sameAddress } from '@/utils/addresses'
+import { isMultisigExecutionInfo } from '@/utils/transaction-guards'
 import { TransactionSummary, TransactionStatus } from '@gnosis.pm/safe-react-gateway-sdk'
+import { useMemo } from 'react'
+import useWallet from './wallets/useWallet'
 
 type TxLocalStatus = TransactionStatus | PendingStatus
 
-const STATUS_LABELS: Record<TxLocalStatus, string> = {
-  [TransactionStatus.AWAITING_CONFIRMATIONS]: 'Awaiting confirmations',
-  [TransactionStatus.AWAITING_EXECUTION]: 'Awaiting execution',
-  [TransactionStatus.CANCELLED]: 'Cancelled',
-  [TransactionStatus.FAILED]: 'Failed',
-  [TransactionStatus.SUCCESS]: 'Success',
-  [PendingStatus.SUBMITTING]: 'Submitting',
-  [PendingStatus.MINING]: 'Mining',
-  [PendingStatus.INDEXING]: 'Indexing',
+const hasSignature = (executionInfo: TransactionSummary['executionInfo'], address?: string) => {
+  return isMultisigExecutionInfo(executionInfo)
+    ? executionInfo.missingSigners?.some((signer) => sameAddress(signer.value, address))
+    : false
 }
```

Comment:
This looks very similar to `isSignableBy`. Can we reuse that instead?

Reply:
We sure can. Nice catch!